plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/MachineParameters.hs

```diff
@@ -22,10 +21,10 @@ import           GHC.Types                                      (Type)
    internal costs, so we keep the machine costs abstract.  The model for Cek
    machine steps is in UntypedPlutusCore.Evaluation.Machine.Cek.CekMachineCosts.
 -}
-data CostModel machinecosts =
+data CostModel machinecosts builtincosts =
     CostModel {
       machineCostModel :: machinecosts
-    , builtinCostModel :: BuiltinCostModel
+    , builtinCostModel :: builtincosts
     } deriving (Eq, Show)
```

Comment:
> @kwxm I've made `CostModel` extensible

Could you check that `plutus-ledger-api` still builds OK and that the tests pass?  `Api.hs` in `plutus-ledger-api` is used by the ledger people and it uses the stuff in `CostModelInterface`, so if anything goes wrong there it could case problems. (I'm off this week so I haven't looked closely at your changes.)

Reply:
... and if you want to change that thing that you mentioned earlier about `term` in `MachineCosts`, please go ahead (if it still applies).

Reply:
> (I'm off this week so I haven't looked closely at your changes.)

They're a super straightforward generalization, so shouldn't be problematic at all, just notifying you of the change.

> Could you check that `plutus-ledger-api` still builds OK and that the tests pass?

The CI wouldn't let us merge the commit if `plutus-ledger-api` tests were not OK, right? Anyway, I checked locally and they pass. Interestingly, I didn't need to update anything outside of Plutus Core...

> ... and if you want to change that thing that you mentioned earlier about term in `MachineCosts`, please go ahead (if it still applies).

I think I don't care enough, but maybe I will.

Reply:
Ah, yeah, potentially that means you could move `NoOp1` etc outside into `ExtensionFun` where all the weird non-production builtins live, but maybe it's simpler to keep those in `DefaultFun`.

Reply:
> Ah, yeah, potentially that means you could move NoOp1 etc outside into ExtensionFun where all the weird non-production builtins live, but maybe it's simpler to keep those in DefaultFun.

That would probably be a good idea.  I just stuck them in `DefaultFun` because I was in a hurry.  There's an issue in JIRA to do that: https://jira.iohk.io/browse/SCP-2267.